File: backend/app/models/time_slot.py
from sqlalchemy import Column, Integer, Time
from app.database import Base

class TimeSlot(Base):
    __tablename__ = "time_slots"

    id = Column(Integer, primary_key=True, index=True)
    start_time = Column(Time, nullable=False)
    end_time = Column(Time, nullable=False)
    slot_order = Column(Integer, nullable=False)

File: SmartTimetable/src/SmartTimetable.Core/Entities/TimeSlot.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTimetable.Core.Entities;

[Table("time_slots")]
public class TimeSlot
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("start_time")]
    public TimeSpan StartTime { get; set; }

    [Required]
    [Column("end_time")]
    public TimeSpan EndTime { get; set; }

    [Required]
    [Column("slot_order")]
    public int SlotOrder { get; set; }
}
